internal/ansible/dispatcher/dispatcher_events_test.go

```diff
@@ -0,0 +1,375 @@
+			3,
+			&host,
+			playbookJSONResult.Tasks[0].Task,
+			playbookJSONResult.Tasks[0].Hosts[host],
+			taskUUIDs,
+			duplicates)
+
+		expectedJson = replaceIDs(expectedJson, msgPlaybookOnStart.Uuid, *msgPlaybookOnStart.EventData.PlaybookUuid)
+
+		msgList := ansibleDispatcher.AddRunnerJobEvent(*msgPlaybookOnStart)
+
+		//When
+		jsonBytes, err := dispatcher.CreateMessage(msgList)
+
+		//Expect
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(string(jsonBytes)).To(Equal(expectedJson))
+	})
+	It("Playbook on runner ok", func() {
+		//Given
+		expectedJson :=
+			`{	"counter":4,
+				"created":"2022-02-03T08:46:38.412345Z",
+				"end_line":0,
+				"event":"v2_runner_on_ok",
+				"event_data":{
+					"host":"host1",
+					"playbook":"test-playbook-filename.yml",
+					"playbook_uuid":"REPLACE_PLAYBOOK_UUID",
+					"res":{"changed":true},
+					"task":"task-test-name-1",
+					"task_action":"",
+					"task_uuid":"9k7b6825-cf3d-4653-9010-9f75d9358c54",
+					"uuid": "9k7b6825-cf3d-4653-9010-9f75d9358c54"
+				},
+				"runner_ident":"test-device-id-12345",
+				"start_line":0,
+				"stdout":"",
+				"uuid":"REPLACE_UUID"}`
+
+		msgRunnerOk := ansibleDispatcher.RunnerOnOk(
+			deviceId,
+			playbookFilename,
+			playbookUuid,
+			4,
+			&host,
+			playbookJSONResult.Tasks[0].Task,
+			playbookJSONResult.Tasks[0].Hosts[host])
+
+		expectedJson = replaceIDs(expectedJson, msgRunnerOk.Uuid, *msgRunnerOk.EventData.PlaybookUuid)
+
+		msgList := []message.AnsibleRunnerJobEventYaml{*msgRunnerOk}
+
+		//When
+		jsonBytes, err := dispatcher.CreateMessage(msgList)
+
+		//Expect
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(string(jsonBytes)).To(Equal(expectedJson))
+	})
+	It("Playbook on runner failed", func() {
+		//Given
+		expectedJson :=
+			`{	"counter":4,
+				"created":"2022-02-03T08:46:38.412345Z",
+				"end_line":0,
+				"event":"v2_runner_on_failed",
+				"event_data":{
+					"host":"host1",
+					"playbook":"test-playbook-filename.yml",
+					"playbook_uuid":"REPLACE_PLAYBOOK_UUID",
+					"res":{"changed":true},
+					"task":"task-test-name-1",
+					"task_action":"",
+					"task_uuid":"9k7b6825-cf3d-4653-9010-9f75d9358c54",
+					"uuid": "9k7b6825-cf3d-4653-9010-9f75d9358c54"
+				},
+				"runner_ident":"test-device-id-12345",
+				"start_line":0,
+				"stderr":"",
+				"stdout":"",
+				"uuid":"REPLACE_UUID"}`
+
+		msgRunnerFailed := ansibleDispatcher.RunnerOnFailed(
+			deviceId,
+			playbookFilename,
+			playbookUuid,
+			4,
+			&host,
+			playbookJSONResult.Tasks[0].Task,
+			playbookJSONResult.Tasks[0].Hosts[host])
+
+		expectedJson = replaceIDs(expectedJson, msgRunnerFailed.Uuid, *msgRunnerFailed.EventData.PlaybookUuid)
+
+		msgList := []message.AnsibleRunnerJobEventYaml{*msgRunnerFailed}
+
+		//When
+		jsonBytes, err := dispatcher.CreateMessage(msgList)
+
+		//Expect
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(string(jsonBytes)).To(Equal(expectedJson))
+	})
+	It("Playbook on runner skipped", func() {
+		//Given
+		expectedJson :=
```

Comment:
This json is very similar, and after that you do replace, a good thing is to use a go struct, have something like


`get_expected_json := func(UUID, counter) []bytes` and it's a bit clear, but keep like this, this PR is already big!

Reply:
I know it isn't ideal, but if even the differences among expectedJson strings are small... there are still there. 
I avoided factoring the code further hoping to make the test more talky. But maybe I didn't succeed!
Maybe we can improve this in another PR.